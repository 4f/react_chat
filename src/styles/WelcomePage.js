export default theme => ({
  paper: {
    marginTop: 64 + theme.spacing.unit * 3,
    width: 500
  },
  tabContent: {
    padding: theme.spacing.unit * 3
  },
  tab: {
    '&:hover': {
      background: 'yellow'
    }
  }
})
