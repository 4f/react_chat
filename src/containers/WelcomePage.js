import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { signup, login, recieveAuth } from 'actions'
import { withStyles } from 'material-ui'
import Welcome from 'components/welcome'
import styles from 'styles/WelcomePage'

const mapStateToProps = state => ({
  isAuth: state.auth.isAuth
})

const mapDispatchToProps = dispatch => bindActionCreators({
  signup,
  login,
  recieveAuth
}, dispatch)


export default connect(
  mapStateToProps,
  mapDispatchToProps
)( withStyles(styles)(Welcome) )
