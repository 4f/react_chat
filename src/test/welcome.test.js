/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import WelcomePage from 'components/welcome';

jest.mock('components/welcome/form', () => () => 'Form');

const mockProps = {
  actions: {
    auth: {
      signup: jest.fn(),
      login: jest.fn(),
    },
  },
  classes: {
    tab: 'tab',
    paper: 'paper',
    tabContent: 'tabContent',
  },
};

describe('<WelcomePage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter >
        <WelcomePage {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <WelcomePage {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
