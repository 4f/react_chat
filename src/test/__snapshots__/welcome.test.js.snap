// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage /> renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-195 MuiPaper-shadow4-201 MuiAppBar-root-187 MuiAppBar-positionFixed-188 MuiAppBar-colorPrimary-193 mui-fixed"
>
    <div
        className="MuiToolbar-root-222 MuiToolbar-gutters-223"
    >
        <h2
            className="MuiTypography-root-224 MuiTypography-title-230 MuiTypography-colorInherit-243"
            style={
                Object {
                    "flex": 1,
                  }
            }
        >
            DogeCodes React Chat
        </h2>
    </div>
</header>,
  <div
    className="MuiGrid-typeContainer-248 MuiGrid-justify-xs-center-265"
>
    <div
        className="MuiGrid-typeItem-249"
    >
        <div
            className="MuiPaper-root-195 MuiPaper-shadow2-199 MuiPaper-rounded-196 paper"
        >
            <header
                className="MuiPaper-root-195 MuiPaper-shadow4-201 MuiAppBar-root-187 MuiAppBar-positionStatic-191 MuiAppBar-colorDefault-192"
            >
                <div
                    className="MuiTabs-root-338"
                >
                    <div
                        className="MuiTabs-flexContainer-339"
                    >
                        <div
                            className="MuiTabs-scrollingContainer-340 MuiTabs-fixed-341"
                            onScroll={[Function]}
                            role="tablist"
                            style={
                                Object {
                                    "marginBottom": 0,
                                  }
                            }
                        >
                            <div
                                className="MuiTabs-flexContainer-339"
                            >
                                <button
                                    aria-selected={true}
                                    className="MuiButtonBase-root-361 MuiTab-root-345 MuiTab-rootInherit-347 MuiTab-rootInheritSelected-354 MuiTab-fullWidth-356 tab"
                                    disabled={false}
                                    onBlur={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    onKeyUp={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchMove={[Function]}
                                    onTouchStart={[Function]}
                                    role="tab"
                                    style={undefined}
                                    tabIndex={0}
                                    type="button"
                                >
                                    <span
                                        className="MuiTab-wrapper-357"
                                    >
                                        <span
                                            className="MuiTab-labelContainer-358"
                                        >
                                            <span
                                                className="MuiTab-label-359"
                                            >
                                                Login
                                            </span>
                                        </span>
                                    </span>
                                    <span
                                        className="MuiTouchRipple-root-366"
                                    />
                                </button>
                                <button
                                    aria-selected={false}
                                    className="MuiButtonBase-root-361 MuiTab-root-345 MuiTab-rootInherit-347 MuiTab-fullWidth-356 tab"
                                    disabled={false}
                                    onBlur={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    onKeyUp={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchMove={[Function]}
                                    onTouchStart={[Function]}
                                    role="tab"
                                    style={undefined}
                                    tabIndex={0}
                                    type="button"
                                >
                                    <span
                                        className="MuiTab-wrapper-357"
                                    >
                                        <span
                                            className="MuiTab-labelContainer-358"
                                        >
                                            <span
                                                className="MuiTab-label-359"
                                            >
                                                Sign Up
                                            </span>
                                        </span>
                                    </span>
                                    <span
                                        className="MuiTouchRipple-root-366"
                                    />
                                </button>
                            </div>
                            <span
                                className="MuiTabIndicator-root-363 MuiTabIndicator-colorSecondary-365"
                                style={
                                    Object {
                                        "left": 0,
                                        "width": 0,
                                      }
                                }
                            />
                        </div>
                    </div>
                </div>
            </header>
            <div
                className="tabContent"
            >
                Form
            </div>
        </div>
    </div>
</div>,
]
`;
